sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -279,36 +266,23 @@ scatter(Tx *p, Toffset offset, simd<Tx, N> vals, simd_mask<N> mask = 1) {
 /// the loaded data as a vector. Actual code generated depends on the
 /// alignment parameter.
 /// @tparam Tx Element type.
-/// @tparam N Number of elements to load, <code>N * sizeof(Tx)</code> must be
-///    1, 2, 4 or 8 owords long.
+/// @tparam N Number of elements to load.
 /// @tparam Flags The alignment specifier type tag. Auto-deduced from the
 ///    \c Flags parameter. If it is less than \c 16, then slower unaligned
 ///    access is generated, otherwise the access is aligned.
 /// @param addr The address to load from.
 /// @param Flags Specifies the alignment.
 /// @return A vector of loaded elements.
 ///
-template <typename Tx, int N, typename Flags = vector_aligned_tag,
-          class T = detail::__raw_t<Tx>,
-          typename = std::enable_if_t<is_simd_flag_type_v<Flags>>>
-__ESIMD_API simd<Tx, N> block_load(const Tx *addr, Flags = {}) {
-  constexpr unsigned Sz = sizeof(T) * N;
-  static_assert(Sz >= detail::OperandSize::OWORD,
```

Comment:
it's really cool we can remove all these restrictions, but are we sure the hardware can support them? i assume IGC lowers the loads to the same hardware instructions as the VC intrinsic we used before, so i would think the same restrictions apply, but i am not a hardware expert.

Reply:
I extended the tests.
IGC lowers VC intrinsics and LLVM-IR load/store instructions differently.
VC-intrinsics are lowered mostly 1-to-1.

Reply:
oh ok, interesting